Add preliminary documentation for okta input (#860)
Graylog2/documentation: pages/integrations/inputs/okta_input.rst

.. _okta_input:

*********************
Okta Log Events Input
*********************

> .. attention:: This is an Enterprise Integrations feature and is only available since Graylog version 3.3 and thus requires an Enterprise license.

Reviewer: ```suggestion
.. attention:: This is a Graylog Enterprise feature and is only available since Graylog version 3.3.  A valid Graylog Enterprise license is required.
```

> This input allows Graylog to pull Okta Log Events from your okta domain.

Reviewer: ```suggestion
This input allows Graylog to pull event logs from your Okta domain.
```

> For this input plugin to function as expected, the following items need to be supplied in the input module.

Reviewer: ```suggestion
For this input plugin to function as expected, the following items must be supplied in the input configuration:
```

1) Domain name

>     The domain name associated with the IP address of the inbound event request (where the Okta Log Events are located).

Reviewer: ```suggestion
    Your Okta Domain (also known as your Okta URL).  You can copy your domain from the Okta Developer Console.
```

Reviewer: Also consider adding a link to https://developer.okta.com/docs/guides/find-your-domain/overview/

2) API Key

>     The API token that allows for polling Okta Log Events.

Reviewer: ```suggestion
    The token which will be used to authenticate Graylog's requests to Okta.  You can create an API token on the Okta Developer Console.
```

Also consider adding a link to https://developer.okta.com/docs/guides/create-an-api-token/-/overview/

> 3) Pull Log Event Since

Reviewer: Need to add information about the Polling Interval.  Determines how often we will request new data from Okta.  Cannot be smaller than 5 seconds.

    The lower time bound of the Okta log events. Determines how much historical data Graylog will pull from Okta when the Input starts.
    If not provided, 1 polling interval of historical data will be pulled.  Must be a timestamp in ISO-8601 format."

> 4) Keyword filter

Reviewer: Should note that the Keyword filter is optional.

If it is provided, cannot have more than 10 keywords (space-separated) and keywords cannot be more than 40 characters long.

.. _okta_input:

*********************
Okta Log Events Input
*********************

.. attention:: This is a Graylog Enterprise feature and is only available since Graylog version 3.3. A valid Graylog Enterprise license is required.

This input allows Graylog to pull event logs from your Okta domain.

For this input plugin to function as expected, the following items must be supplied in the input configuration:

1) Domain name
    Your Okta Domain (also known as your Okta URL). You can copy your domain from the Okta Developer Console.
    For additional information see: https://developer.okta.com/docs/guides/find-your-domain/overview/

2) API key
    The token which will be used to authenticate Graylog's requests to Okta.
    You can create an API token on the Okta Developer Console.
    For additional information see: https://developer.okta.com/docs/guides/create-an-api-token/-/overview/

3) Pull log events since

>     The lower time bound of the Okta log events and determines how much historical data Graylog will pull from Okta when the Input starts.

Reviewer 2: Please add a link, to the api link. 
https://developer.okta.com/docs/reference/api/system-log/ 
The document looks good.

.. _okta_input:

*********************
Okta Log Events Input
*********************

.. attention:: This is a Graylog Enterprise feature and is only available since Graylog version 3.3. A valid Graylog Enterprise license is required.

> This input allows Graylog to pull event logs from your Okta domain. For an example of an Okta Log Event object see: https://developer.okta.com/docs/reference/api/system-log/#logevent-object

Reviewer 3: > The Okta System Log records system events related to your organization in order to provide an audit trail that can be used to understand platform activity and to diagnose problems.

The above is taken from the referenced page - maybe it would make sense to integrate that too? 

Why someone should install this plugin? What benefit comes out of that? Not in marketing speak - from a technical point of view.

Reviewer 2: **What is Okta ?***

Helps organization with 2 things. 

![okta_solutions](https://user-images.githubusercontent.com/6947168/80851397-2accea00-8be7-11ea-9763-73eb1a19e7e7.png)

**Why graylog Okta Plugin ?** 
* launch the `okta Log Events` plugin to ingest okta syslog and analyze them. 
* [tell stories, that are insightful, inspiring , innovate,change the way you think, with a custom dashboard.](https://github.com/Graylog2/documentation/blob/3.2/pages/dashboards.rst#thats-it)
* set up [alerts](https://github.com/Graylog2/documentation/blob/3.2/pages/alerts.rst#id2) with a combination of fields, saving millions of dollars on down time events, save the earth.  
* pay attention to the odd-balls in your data.
* refer @waab76 notes from the all hans demo.

1) Domain name
    Your Okta Domain (also known as your Okta URL). You can copy your domain from the Okta Developer Console.
    For additional information see: https://developer.okta.com/docs/guides/find-your-domain/overview/

2) API key
    The token which will be used to authenticate Graylog's requests to Okta.
    You can create an API token on the Okta Developer Console.
    For additional information see: https://developer.okta.com/docs/guides/create-an-api-token/-/overview/

3) Pull log events since
    The lower time bound of the Okta log events and determines how much historical data Graylog will pull from Okta when the Input starts.
    If not provided, 1 polling interval of historical data will be pulled.  Must be a timestamp in ISO-8601 format."

4) Polling interval
    Determines how often Graylog will poll for new data stored in Okta. Cannot be smaller than 5 seconds.

5) Keyword filter (optional)
    The keyword filter is optional and is used to filter the log events results.
    If it is provided, then it cannot have more than 10 keywords (space-separated) and keywords cannot be more than 40 characters long.

> .. Note:: It is recommended that this input only be run on one designated node. Selecting Global will not ensure that the input is running on master, as well as may result in duplicate data.

Reviewer 2: thanks for adding this line. it will ensure, they don't check the global checkbox, we didn't find a way to remove the `isGlobal` check box yet.

Reviewer 3: This is better than nothing. But this will be something that came up regular. Will it be able to prevent that more than one input is running? Could we implement something in some way?

Reviewer 4: updated the verbiage. if global is selected, then the input will default to only running on the master node
